script.js

```diff
@@ -1,4 +1,210 @@
+    console.log("pushed player1secondrolloutput into player1array");
+
+    console.log(player1Array);
+
+    currentGameMode = player1Select; //change game mode as soon as dices are rolled so it won't re-roll
+    console.log("game mode is changed to player1select");
+
+    myOutputValue = `<b>WELCOME PLAYER 1 </b>🎲🎲🎲 <br><br>You rolled ${player1FirstRollOutput} for Dice 1 and ${player1SecondRollOutput} for Dice 2.<br><br>${selectDiceOrderText} `;
+    return myOutputValue;
+  }
+
+  //game mode: player 1 to select which dice first
+  if (currentGameMode == player1Select) {
+    // how to collect user choice of which dice goes first? (kiv)
+    userInput = Number(input);
+    var player2TurnText = "It is now Player 2's turn. Click submit to proceed.";
+
+    //input validation for (1) nan inputs (2) not 1 and not 2
+
+    //for (1) nan inputs
+    if (Number.isNaN(userInput) == true) {
+      myOutputValue = "sorry, please enter a number.";
+      return myOutputValue;
+    }
+
+    if (userInput != 1 && userInput != 2) {
+      myOutputValue = "Please insert only either '1' or '2'.";
+      return myOutputValue;
+    }
+
+    if (userInput == 1) {
+      //if input is 1, let the first number go first. Else if input is 2, let second number go first
+
+      player1Number = String(player1Array[0]) + String(player1Array[1]);
+      myOutputValue = `Player 1, you chose Dice ${userInput} first.<br><br>You rolled ${player1Array}.<br><b>Your number is ${player1Number}.</b><br><br>${player2TurnText}`;
+      currentGameMode = player2Roll; // change game mode should be within conditionals
+      console.log("game mode is changed to player2roll");
+      return myOutputValue;
+    } else if (userInput == 2) {
+      player1Number = String(player1Array[1]) + String(player1Array[0]);
+      myOutputValue = `Player 1, you chose Dice ${userInput} first.<br><br>You rolled ${player1Array}.<br><b>Your number is ${player1Number}.</b><br><br>${player2TurnText} `;
+      currentGameMode = player2Roll; // change game mode should be within conditionals
+      console.log("game mode is changed to player2roll");
+
+      return myOutputValue;
+    }
+  }
+
+  //game mode: player 2 to roll
+  if (currentGameMode == player2Roll) {
+    player2FirstRollOutput = diceRoll();
+    player2SecondRollOutput = diceRoll();
+    player2Array.push(player2FirstRollOutput);
+    player2Array.push(player2SecondRollOutput);
+
+    currentGameMode = player2Select;
+    console.log("game mode is changed to player2select");
+    myOutputValue = `<b>WELCOME PLAYER 2 </b>🎲🎲🎲<br><br>You rolled ${player2FirstRollOutput} for Dice 1 and ${player2SecondRollOutput} for Dice 2.<br><br>${selectDiceOrderText}`;
+    return myOutputValue;
+  }
+  // game mode: player 2 to select
+
+  userInput = Number(input);
+  var textToProceedToCheckWinner = "Click submit to find out who won!🧐🤔";
+
+  if (currentGameMode == player2Select) {
+    //input validation for (1) not a number (2) not 1 and not 2
```

Comment:
the code for this game mode and the previous (player2Roll) is almost entirely similar to the code for game modes `player1Roll` and `player1Select`, other than just the change in player and the addition of your `textToProceedToCheckWinner`. consider how you can abstract these into its own functions, maybe say `playerRoll` and `playerSelect`. You can then have a global variable called `curPlayer` which tracks who the current player is. This way you can customise your output and the array you calculate the score from based on this global variable.

This one is a bit complicated for me to show an example but do give it a shot and let me know if you face issues! This would clean up your code significantly by 1. reducing repetition and 2. separating different aspects of the gameplay into its own function which makes your `beatThatGame` function shorter. One thing you wanted specific feedback on is how to refactor code so this would definitely be an area that I would start with!